shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java

```diff
@@ -53,16 +57,66 @@ public static void addMetaTag(
     }
 
     public static void setMessageTypeCoding(Bundle order, Coding coding) {
-        var messageHeader = findOrInitializeMessageHeader(order);
+        var messageHeader = findOrCreateMessageHeader(order);
         messageHeader.setEvent(coding);
     }
 
-    public static MessageHeader findOrInitializeMessageHeader(Bundle bundle) {
+    public static MessageHeader findOrCreateMessageHeader(Bundle bundle) {
         var messageHeader = resourceInBundle(bundle, MessageHeader.class);
         if (messageHeader == null) {
             messageHeader = new MessageHeader();
             bundle.addEntry(new Bundle.BundleEntryComponent().setResource(messageHeader));
         }
         return (MessageHeader) messageHeader;
     }
+
+    public static void addSendingFacilityToMessageHeader(Bundle bundle, String name) {
+        var header = findOrCreateMessageHeader(bundle);
+        var org = new Organization();
+        org.setName(name);
+        header.setSender(new Reference(org));
+    }
+
+    public static void addReceivingFacilityToMessageHeader(Bundle bundle, String name) {
+        var header = findOrCreateMessageHeader(bundle);
+        var org = new Organization();
+        org.setName(name);
+        var destination = new MessageHeader.MessageDestinationComponent();
+        destination.setReceiver(new Reference(org));
+        header.setDestination(Collections.singletonList(destination));
+        bundle.addEntry(new Bundle.BundleEntryComponent().setResource(org));
+    }
+
+    public static void swapOBRsRenameThis(Bundle bundle) {
```

Comment:
@luis-pabon-tf  I think this code should go in the custom transform method as it's specific to a transformation. The HapiHelper library should have more generic functions that can be used by multiple transformations